<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content=" Guangchuang Yu (guangchuangyu@gmail.com) and Tommy Tsan-Yuk Lam (ttylam@hku.hk) School of Public Health, The University of Hong Kong" />

<meta name="date" content="2015-12-29" />

<title>ggtree: a phylogenetic tree viewer for different types of tree annotations</title>


<style type="text/css">code{white-space: pre;}</style>
<link href="data:text/css;charset=utf-8,pre%20%2Eoperator%2C%0Apre%20%2Eparen%7Bcolor%3Argb%28104%2C%20118%2C%20135%29%7Dpre%20%2Eliteral%7Bcolor%3A%23990073%7Dpre%20%2Enumber%7Bcolor%3A%23099%7Dpre%20%2Ecomment%7Bcolor%3A%23998%3Bfont%2Dstyle%3Aitalic%7Dpre%20%2Ekeyword%7Bcolor%3A%23900%3Bfont%2Dweight%3Abold%7Dpre%20%2Eidentifier%7Bcolor%3Argb%280%2C%200%2C%200%29%7Dpre%20%2Estring%7Bcolor%3A%23d14%7D" rel="stylesheet" type="text/css" />
<script src="data:application/x-javascript;base64,
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();

"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>


<link href="data:text/css;charset=utf-8,body%2C%20td%7Bfont%2Dfamily%3Asans%2Dserif%3Bbackground%2Dcolor%3Awhite%3Bfont%2Dsize%3A13px%7Dbody%7Bmax%2Dwidth%3A800px%3Bmargin%3A0%20auto%3Bpadding%3A1em%201em%202em%3Bline%2Dheight%3A20px%7Ddiv%23TOC%20li%7Blist%2Dstyle%3Anone%3Bbackground%2Dimage%3Anone%3Bbackground%2Drepeat%3Anone%3Bbackground%2Dposition%3A0%7Dp%2C%20pre%7Bmargin%3A0em%200em%201em%7Dimg%2C%20table%7Bmargin%3A0em%20auto%201em%7Dp%7Btext%2Dalign%3Ajustify%7Dtt%2C%20code%2C%20pre%7Bfont%2Dfamily%3A%27DejaVu%20Sans%20Mono%27%2C%20%27Droid%20Sans%20Mono%27%2C%20%27Lucida%20Console%27%2C%20Consolas%2C%20Monaco%2C%20monospace%7Dh1%2C%20h2%2C%20h3%2C%20h4%2C%20h5%2C%20h6%7Bfont%2Dfamily%3AHelvetica%2C%20Arial%2C%20sans%2Dserif%3Bmargin%3A1%2E2em%200em%200%2E6em%200em%3Bfont%2Dweight%3Abold%7Dh1%2Etitle%7Bfont%2Dsize%3A250%25%3Bfont%2Dweight%3Anormal%3Bcolor%3A%2387b13f%3Bline%2Dheight%3A1%2E1em%3Bmargin%2Dtop%3A0px%3Bborder%2Dbottom%3A0px%7Dh1%7Bfont%2Dsize%3A160%25%3Bfont%2Dweight%3Anormal%3Bline%2Dheight%3A1%2E4em%3Bborder%2Dbottom%3A1px%20%231a81c2%20solid%7Dh2%7Bfont%2Dsize%3A130%25%7Dh1%2C%20h2%2C%20h3%7Bcolor%3A%231a81c2%7Dh3%2C%20h4%2C%20h5%2C%20h6%7Bfont%2Dsize%3A115%25%7Da%7Bcolor%3A%231a81c2%7Da%3Aactive%7Boutline%3Anone%7Da%3Avisited%7Bcolor%3A%231a81c2%7Da%3Ahover%7Bcolor%3A%234c94c2%7Dpre%2C%20img%7Bmax%2Dwidth%3A100%25%3Bdisplay%3Ablock%7Dpre%7Bborder%3A0px%20none%3Bbackground%2Dcolor%3A%23F8F8F8%3Bwhite%2Dspace%3Apre%3Boverflow%2Dx%3Aauto%7Dpre%20code%7Bborder%3A1px%20%23aaa%20dashed%3Bbackground%2Dcolor%3Awhite%3Bdisplay%3Ablock%3Bpadding%3A1em%3Bcolor%3A%23111%3Boverflow%2Dx%3Ainherit%7Dpre%20code%5Bclass%5D%7Bbackground%2Dcolor%3Ainherit%7Dpre%5Bclass%5D%20code%7Bbackground%2Dcolor%3Ainherit%7Dcode%7Bbackground%2Dcolor%3Atransparent%3Bcolor%3A%2387b13f%3Bfont%2Dsize%3A92%25%7Dtable%2C%20td%2C%20th%7Bborder%3Anone%3Bpadding%3A0%200%2E5em%7Dtbody%20tr%3Anth%2Dchild%28odd%29%20td%7Bbackground%2Dcolor%3A%23F8F8F8%7Dblockquote%7Bcolor%3A%23666666%3Bmargin%3A0%3Bpadding%2Dleft%3A1em%3Bborder%2Dleft%3A0%2E5em%20%23EEE%20solid%7Dhr%7Bheight%3A0px%3Bborder%2Dbottom%3Anone%3Bborder%2Dtop%2Dwidth%3Athin%3Bborder%2Dtop%2Dstyle%3Adotted%3Bborder%2Dtop%2Dcolor%3A%23999999%7Dspan%2Eheader%2Dsection%2Dnumber%7Bpadding%2Dright%3A1em%7Dspan%2Etoc%2Dsection%2Dnumber%3A%3Aafter%7Bcontent%3A%22%20%20%22%3Bwhite%2Dspace%3Apre%7D%40media%20print%7B%2A%7Bbackground%3Atransparent%20%21important%3Bcolor%3Ablack%20%21important%3Bfilter%3Anone%20%21important%3B%2Dms%2Dfilter%3Anone%20%21important%7Dbody%7Bfont%2Dsize%3A12pt%3Bmax%2Dwidth%3A100%25%7Da%2C%20a%3Avisited%7Btext%2Ddecoration%3Aunderline%7Dhr%7Bvisibility%3Ahidden%3Bpage%2Dbreak%2Dbefore%3Aalways%7Dpre%2C%20blockquote%7Bpadding%2Dright%3A1em%3Bpage%2Dbreak%2Dinside%3Aavoid%7Dtr%2C%20img%7Bpage%2Dbreak%2Dinside%3Aavoid%7Dimg%7Bmax%2Dwidth%3A100%25%20%21important%7D%40page%20%3Aleft%7Bmargin%3A15mm%2020mm%2015mm%2010mm%7D%40page%20%3Aright%7Bmargin%3A15mm%2010mm%2015mm%2020mm%7Dp%2C%20h2%2C%20h3%7Borphans%3A3%3Bwidows%3A3%7Dh2%2C%20h3%7Bpage%2Dbreak%2Dafter%3Aavoid%7D%7D" rel="stylesheet" type="text/css" />

<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if(links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script>

</head>

<body>


<div id="header">
<h1 class="title">ggtree: a phylogenetic tree viewer for different types of tree annotations</h1>
<h4 class="author"><em><br />
Guangchuang Yu (<a href="mailto:guangchuangyu@gmail.com">guangchuangyu@gmail.com</a>) and Tommy Tsan-Yuk Lam (<a href="mailto:ttylam@hku.hk">ttylam@hku.hk</a>)<br />
School of Public Health, The University of Hong Kong</em></h4>
<h4 class="date"><em>2015-12-29</em></h4>
</div>

<h1>Contents</h1>
<div id="TOC">
<ul>
<li><a href="#citation"><span class="toc-section-number">1</span> Citation</a></li>
<li><a href="#introduction"><span class="toc-section-number">2</span> Introduction</a></li>
<li><a href="#getting-data-into-r"><span class="toc-section-number">3</span> Getting data into <code>R</code></a></li>
<li><a href="#tree-visualization-and-annotation"><span class="toc-section-number">4</span> Tree Visualization and Annotation</a></li>
<li><a href="#vignette-entry"><span class="toc-section-number">5</span> Vignette Entry</a></li>
<li><a href="#bugsfeature-requests"><span class="toc-section-number">6</span> Bugs/Feature requests</a></li>
<li><a href="#session-info"><span class="toc-section-number">7</span> Session info</a></li>
</ul>
</div>

<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script>
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script>
<blockquote>
<p>You can’t even begin to understand biology, you can’t understand life, unless you understand what it’s all there for, how it arose - and that means evolution. — Richard Dawkins</p>
</blockquote>
<div id="citation" class="section level1">
<h1><span class="header-section-number">1</span> Citation</h1>
<p>If you use <em><a href="http://bioconductor.org/packages/release/bioc/html/ggtree.html">ggtree</a></em> in published research, please cite:</p>
<pre><code>G Yu, D Smith, H Zhu, Y Guan, TTY Lam,
ggtree: an R package for visualization and annotation of phylogenetic tree with different types of meta-data.
revised.</code></pre>
</div>
<div id="introduction" class="section level1">
<h1><span class="header-section-number">2</span> Introduction</h1>
<p>This project arose from our needs to annotate nucleotide substitutions in the phylogenetic tree, and we found that there is no tree visualization software can do this easily. Existing tree viewers are designed for displaying phylogenetic tree, but not annotating it. Although some tree viewers can displaying bootstrap values in the tree, it is hard/impossible to display other information in the tree. Our first solution for displaying nucleotide substituitions in the tree is to add this information in the node/tip names and use traditional tree viewer to show it. We displayed the information in the tree successfully, but we believe this indirect approach is inefficient.</p>
<p>In the old day, phylogenetic tree is often small. At that time, as we almost didn’t have a need to annotate a tree; displaying the evolution relationships is mostly enough. Nowadays, we can obtain a lot of data from different experiments, and we want to associate our data, for instance antigenic change, with the evolution relationship. Visualizing these associations in the phylogenetic tree can help us to identify evolution patterns. We believe we need a next generation tree viewer that should be programmable and extensible. It can view a phylogenetic tree easily as we did with classical software and support adding annotation data in a layer above the tree. This is the objective of developing the <em><a href="https://github.com/GuangchuangYu/ggtree">ggtree</a></em>. Common tasks of annotating a phylogenetic tree should be easy and complicated tasks can be possible to achieve by adding multiple layers of annotation.</p>
<p>The <em><a href="https://github.com/GuangchuangYu/ggtree">ggtree</a></em> is designed by extending the <em><a href="http://cran.fhcrc.org/web/packages/ggplot2/index.html">ggplot2</a></em><span class="citation"><sup><a href="#ref-wickham_ggplot2_2009">1</a></sup></span> package. It is based on the grammar of graphics and takes all the good parts of <em><a href="http://cran.fhcrc.org/web/packages/ggplot2/index.html">ggplot2</a></em>. There are other R packages that implement tree viewer using <em><a href="http://cran.fhcrc.org/web/packages/ggplot2/index.html">ggplot2</a></em>, including <em><a href="http://cran.fhcrc.org/web/packages/OutbreakTools/index.html">OutbreakTools</a></em>, <em><a href="http://bioconductor.org/packages/release/bioc/html/phyloseq.html">phyloseq</a></em><span class="citation"><sup><a href="#ref-mcmurdie_phyloseq_2013">2</a></sup></span> and <em><a href="https://github.com/gjuggler/ggphylo">ggphylo</a></em>; they mostly create complex tree view functions for their specific needs. Internally, these packages interpret a phylogenetic as a collection of <code>lines</code>, which makes it hard to annotate diverse user input that are related to node (taxa). The <em><a href="https://github.com/GuangchuangYu/ggtree">ggtree</a></em> is different to them by interpreting a tree as a collection of <code>taxa</code> and allowing general flexibilities of annotating phylogenetic tree with diverse types of user inputs.</p>
</div>
<div id="getting-data-into-r" class="section level1">
<h1><span class="header-section-number">3</span> Getting data into <code>R</code></h1>
<p>Most of the tree viewer software (including <code>R</code> packages) focus on <code>Newick</code> and <code>Nexus</code> file format, while there are file formats from different evolution analysis software that contain supporting evidences within the file that are ready for annotating a phylogenetic tree. The <em><a href="https://github.com/GuangchuangYu/ggtree">ggtree</a></em> package define several parser functions and <code>S4</code> classes to store statistical evidences inferred by commonly used software packages. It supports several file format, including:</p>
<ul>
<li>Newick (via <code>ape</code>)</li>
<li>Nexus (via <code>ape</code>)</li>
<li>New Hampshire eXtended format (NHX)</li>
<li>Jplace</li>
</ul>
<p>and software output from:</p>
<ul>
<li><a href="http://beast2.org/">BEAST</a><span class="citation"><sup><a href="#ref-bouckaert_beast_2014">3</a></sup></span></li>
<li><a href="http://sco.h-its.org/exelixis/web/software/epa/index.html">EPA</a><span class="citation"><sup><a href="#ref-berger_EPA_2011">4</a></sup></span></li>
<li><a href="http://hyphy.org/w/index.php/Main_Page">HYPHY</a><span class="citation"><sup><a href="#ref-pond_hyphy_2005">5</a></sup></span></li>
<li><a href="http://abacus.gene.ucl.ac.uk/software/paml.html">PAML</a><span class="citation"><sup><a href="#ref-yang_paml_2007">6</a></sup></span></li>
<li><a href="http://pbil.univ-lyon1.fr/software/phyldog/">PHYLDOG</a><span class="citation"><sup><a href="#ref-boussau_genome-scale_2013">7</a></sup></span></li>
<li><a href="http://matsen.fhcrc.org/pplacer/">pplacer</a><span class="citation"><sup><a href="#ref-matsen_pplacer_2010">8</a></sup></span></li>
<li><a href="http://loco.biosci.arizona.edu/r8s/">r8s</a><span class="citation"><sup><a href="#ref-marazzi_locating_2012">9</a></sup></span></li>
<li><a href="http://sco.h-its.org/exelixis/web/software/raxml/">RAxML</a><span class="citation"><sup><a href="#ref-stamatakis_raxml_2014">10</a></sup></span></li>
<li><a href="http://revbayes.github.io/intro.html">RevBayes</a><span class="citation"><sup><a href="#ref-hohna_probabilistic_2014">11</a></sup></span></li>
</ul>
<p>Parsing data from a number of molecular evolution software is not only for visualization in <code>ggtree</code>, but also bring these data to <code>R</code> users for further analysis (e.g. summarization, visualization, comparision, test, <em>etc</em>).</p>
<p>For more details, please refer to <a href="treeImport.html">Tree Data Import</a> vignette.</p>
</div>
<div id="tree-visualization-and-annotation" class="section level1">
<h1><span class="header-section-number">4</span> Tree Visualization and Annotation</h1>
<p>Tree Visualization in <code>ggtree</code> is easy, with one line of command <code>ggtree(tree_object)</code>. It supports several layout, including <code>rectangular</code>, <code>slanted</code> and <code>circular</code> for <code>Phylogram</code> and <code>Cladogram</code>, <code>unrooted</code> layout, time-scaled and two dimentional phylogenies. <a href="treeVisualization.html">Tree Visualization</a> vignette describe these feature in details.</p>
<p>We implement several functions to manipulate a phylogenetic tree.</p>
<ul>
<li>taxa can be clustered together using <code>groupClade</code> or <code>groupOTU</code> functions</li>
<li>clades can be collapsed via <code>collapse</code> function</li>
<li>collapsed clade can be expanded by using <code>expand</code> function</li>
<li>clade can be re-scale to zoom in or zoom out by <code>scaleClade</code> function</li>
<li>selected clade can be rotated by 180 degree using <code>rotate</code> function</li>
<li>position of two selected clades (should share a same parent) can be exchanged by <code>flip</code> function</li>
</ul>
<p>Details and examples can be found in <a href="treeManipulation.html">Tree Manipulation</a> vignette.</p>
<p>Most of the phylogenetic trees are scaled by evolutionary distance (substitution/site), in <code>ggtree</code> a phylogenetic tree can be re-scaled by any numerical variable inferred by evolutionary analysis (e.g. species divergence time, <em>dN/dS</em>, <em>etc</em>). Numerical and category variable can be used to color a phylogenetic tree.</p>
<p>The <code>ggtree</code> package provides several layers to annotate a phylogenetic tree, including <code>geom_treescale</code> for adding a legend of tree scale, <code>geom_hilight</code> for highlighting selected clades and <code>geom_cladelabel</code> for labelling selected clades.</p>
<p>It supports annotating phylogenetic trees with analyses by R packages and other commonly used evolutionary software. User’s specific annotation (e.g. experimental data) can be integrated to annotate phylogenetic tree. <code>ggtree</code> provides <code>write.jplace</code> function to combine Newick tree file and user’s own data to a single <code>jplace</code> file that can be parsed and the data can be used to annotate the tree directly in <code>ggtree</code>.</p>
<p><code>ggtree</code> integrates <code>phylopic</code> database and silhouette images of organisms can be downloaded and used to annotate phylogenetic directly. <code>ggtree</code> also supports using local images to annotate a phylogenetic tree.</p>
<p>Visualizing an annotated phylogenetic tree with numerical matrix (e.g. genotype table) and multiple sequence alignment is also supported in <code>ggtree</code>. Examples of annotating phylogenetic trees can be found in the <a href="treeAnnotation.html">Tree Annotation</a> vignette.</p>
</div>
<div id="vignette-entry" class="section level1">
<h1><span class="header-section-number">5</span> Vignette Entry</h1>
<ul>
<li><a href="treeImport.html">Tree Data Import</a></li>
<li><a href="treeVisualization.html">Tree Visualization</a></li>
<li><a href="treeManipulation.html">Tree Manipulation</a></li>
<li><a href="treeAnnotation.html">Tree Annotation</a></li>
</ul>
</div>
<div id="bugsfeature-requests" class="section level1">
<h1><span class="header-section-number">6</span> Bugs/Feature requests</h1>
<p>If you have any, <a href="https://github.com/GuangchuangYu/ggtree/issues">let me know</a>. Thx!</p>
</div>
<div id="session-info" class="section level1">
<h1><span class="header-section-number">7</span> Session info</h1>
<p>Here is the output of <code>sessionInfo()</code> on the system on which this document was compiled:</p>
<pre><code>## R version 3.2.2 (2015-08-14)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: elementary OS Freya
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=C              
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] phangorn_2.0.1      ggtree_1.3.7        gridExtra_2.0.0    
##  [4] ggplot2_2.0.0       ape_3.4             Biostrings_2.38.2  
##  [7] XVector_0.10.0      IRanges_2.4.6       S4Vectors_0.8.5    
## [10] BiocGenerics_0.16.1 colorspace_1.2-6    BiocStyle_1.8.0    
## 
## loaded via a namespace (and not attached):
##  [1] igraph_1.0.1    Rcpp_0.12.2     knitr_1.11      magrittr_1.5   
##  [5] nnls_1.4        zlibbioc_1.16.0 munsell_0.4.2   lattice_0.20-33
##  [9] quadprog_1.5-5  stringr_1.0.0   plyr_1.8.3      tools_3.2.2    
## [13] grid_3.2.2      nlme_3.1-122    gtable_0.1.2    htmltools_0.2.6
## [17] yaml_2.1.13     digest_0.6.8    Matrix_1.2-3    reshape2_1.4.1 
## [21] formatR_1.2.1   evaluate_0.8    rmarkdown_0.9   stringi_1.0-1  
## [25] scales_0.3.0    jsonlite_0.9.19</code></pre>
<div class="references">
<h1 id="references" class="unnumbered">References</h1>
<div id="ref-wickham_ggplot2_2009">
<p>1. Wickham, H. <em>Ggplot2: Elegant graphics for data analysis</em>. (Springer, 2009).</p>
</div>
<div id="ref-mcmurdie_phyloseq_2013">
<p>2. McMurdie, P. J. &amp; Holmes, S. Phyloseq: An r package for reproducible interactive analysis and graphics of microbiome census data. <em>PLoS ONE</em> <strong>8,</strong> e61217 (2013).</p>
</div>
<div id="ref-bouckaert_beast_2014">
<p>3. Bouckaert, R. <em>et al.</em> BEAST 2: A software platform for bayesian evolutionary analysis. <em>PLoS Comput Biol</em> <strong>10,</strong> e1003537 (2014).</p>
</div>
<div id="ref-berger_EPA_2011">
<p>4. Berger, S. A., Krompass, D. &amp; Stamatakis, A. Performance, accuracy, and web server for evolutionary placement of short sequence reads under maximum likelihood. <em>Systematic Biology</em> <strong>60,</strong> 291–302 (2011).</p>
</div>
<div id="ref-pond_hyphy_2005">
<p>5. Pond, S. L. K., Frost, S. D. W. &amp; Muse, S. V. HyPhy: Hypothesis testing using phylogenies. <em>Bioinformatics</em> <strong>21,</strong> 676–679 (2005).</p>
</div>
<div id="ref-yang_paml_2007">
<p>6. Yang, Z. PAML 4: Phylogenetic analysis by maximum likelihood. <em>Molecular Biology and Evolution</em> <strong>24,</strong> 1586–1591 (2007).</p>
</div>
<div id="ref-boussau_genome-scale_2013">
<p>7. Boussau, B. <em>et al.</em> Genome-scale coestimation of species and gene trees. <em>Genome Res.</em> <strong>23,</strong> 323–330 (2013).</p>
</div>
<div id="ref-matsen_pplacer_2010">
<p>8. Matsen, F. A., Kodner, R. B. &amp; Armbrust, E. V. Pplacer: Linear time maximum-likelihood and bayesian phylogenetic placement of sequences onto a fixed reference tree. <em>BMC Bioinformatics</em> <strong>11,</strong> 538 (2010).</p>
</div>
<div id="ref-marazzi_locating_2012">
<p>9. Marazzi, B. <em>et al.</em> Locating evolutionary precursors on a phylogenetic tree. <em>Evolution</em> <strong>66,</strong> 3918–3930 (2012).</p>
</div>
<div id="ref-stamatakis_raxml_2014">
<p>10. Stamatakis, A. RAxML version 8: A tool for phylogenetic analysis and post-analysis of large phylogenies. <em>Bioinformatics</em> btu033 (2014). <a href="http://doi.org/10.1093/bioinformatics/btu033">doi:10.1093/bioinformatics/btu033</a></p>
</div>
<div id="ref-hohna_probabilistic_2014">
<p>11. Höhna, S. <em>et al.</em> Probabilistic graphical model representation in phylogenetics. <em>Syst Biol</em> <strong>63,</strong> 753–771 (2014).</p>
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
